hw/ip/spi_host/doc/_index.md

```diff
@@ -237,53 +327,176 @@ These settings are all minimum bounds, and delays in the FSM implementation may
+As shown in the following figure, a similar time-ordering scheme applies for Dual- and Quad-mode transfers.
+However many bits of similar significance are packed into multiple parallel `sd` data lines, with the least significant going to `sd[0]`.
+
+{{< wavejson >}}
+{signal: [
+  ["ByteOrder=0",
+  {name: "sd[0]", wave: "x...22334455x...", data: ["d[28]", "d[24]", "d[20]", "d[16]", "d[12]", "d[8]", "d[4]", "d[0]"]},
+  {name: "sd[1]", wave: "x...22334455x...", data: ["d[29]", "d[25]", "d[21]", "d[17]", "d[13]", "d[9]", "d[5]", "d[1]"]},
+  {name: "sd[2]", wave: "x...22334455x...", data: ["d[30]", "d[26]", "d[22]", "d[18]", "d[14]", "d[10]", "d[6]", "d[2]"]},
+  {name: "sd[3]", wave: "x...22334455x...", data: ["d[31]", "d[27]", "d[23]", "d[19]", "d[15]", "d[11]", "d[7]", "d[3]"]},
+],
+   ["ByteOrder=1",
+  {name: "sd[0]", wave: "x...55443322x...", data: ["d[4]", "d[0]", "d[12]", "d[8]", "d[20]", "d[16]", "d[28]", "d[24]"]},
+  {name: "sd[1]", wave: "x...55443322x...", data: ["d[5]", "d[1]", "d[13]", "d[9]", "d[21]", "d[17]", "d[29]", "d[25]"]},
+  {name: "sd[2]", wave: "x...55443322x...", data: ["d[6]", "d[2]", "d[14]", "d[10]", "d[22]", "d[18]", "d[30]", "d[26]"]},
+  {name: "sd[3]", wave: "x...55443322x...", data: ["d[7]", "d[3]", "d[15]", "d[11]", "d[23]", "d[19]", "d[31]", "d[27]"]},
+  ],
+  ],
+  head: {
+   text: "Serial bit ordering for 32-bit data word (d[31:0]), Quad SPI as a Function of the Parameter 'ByteOrder'",
+  },
+  foot: {
+  text: "(Bits are numbered as they appear when loaded into DATA memory window)"
+  }
+}
+{{< /wavejson >}}
 
-If [Pass-through mode](#pass-through-mode) is enabled then the `cs_n` lines are controlled by *neither* the SPI_HOST hardware nor the firmware register.
-In Pass-though mode, control of the `cs_n` lines passes directly to the inter-module port, `pt_cs_ni`.
+### Command Length and Alignment in DATA
 
-### Arbitrary Length Commands
+Even though the {{< regref "DATA" >}} memory window typically accepts 32-bit words, command segments do not need to use all the bytes from every word. 
 
-Typically the length of each command is limited by the maximum number of bits in the {{< regref "COMMAND_0.TXN_CNT_0" >}}, {{< regref "COMMAND_0.RX_CNT_0" >}}, and {{< regref "COMMAND_0.TX1_CNT_0" >}} fields.
-However, a series  of SPI_HOST IP commands can be merged into one long command sequence using the {{< regref "CONFIGOPTS_0.CSAAT_0" >}} field, which leaves the `cs_n` line low after the first SPI_HOST command.
-By raising the `cs_n` line only after the last command has completed, the peripheral will interpret the resulting transmit/receive sequences as one long command.
+For TX (or Bidirectional) segments, any unused bytes from the latest TX FIFO word are simply ignored at the end of a segment.
+For RX (or Bidirectional) segments, if the last few received bytes do not fill an entire DATA word, the partial word will be zero-padded and inserted into the RX FIFO once the segment is completed.
```

Comment:
```suggestion
For RX (or Bidirectional) segments, if the last few received bytes (?) do not fill an entire DATA word, the partial word will be zero-padded and inserted into the RX FIFO once the segment is completed.
```

zero-padded where? 

And really bytes?

Reply:
Yes. Really bytes (3 or fewer). SPI does not support bit-level transactions (or at least I don't know any devices which do).

This sentence has been clarified as:
```
For RX (or Bidirectional) segments, if the last few bytes received do not fill an entire DATA word, the partial word will be zero-padded and inserted into the RX FIFO once the segment is completed.                                                                                                                       
If ByteOrder=1 (the default, Little-Endian case), this padding will fill the unused most-significant bytes of the final RX DATA word, otherwise the padding will fill the unused least-significant bytes.                                                                                                                   
```